spock-specs/src/test/groovy/org/spockframework/smoke/extension/TempDirExtensionSpec.groovy

```diff
@@ -0,0 +1,104 @@
+package org.spockframework.smoke.extension
+
+import org.spockframework.util.Beta
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Stepwise
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.attribute.PosixFilePermissions
+
+/**
+ * @author dqyuan
+ */
+@Beta
+@Stepwise
+class TempDirExtensionSpec extends Specification {
+
+  @Shared
+  @TempDir
+  Path sharedDir
+  static Path preShared
+
+  @TempDir
+  File iterationDir
+  static File preIteration
+
+  @TempDir(baseDir = "build")
+  Path customerTempDir
```

Comment:
As you have four fields, it would make sense to have one as `Object` and one as `def` to have also covered that it works properly and they get `Path` instances injected.